### PR TITLE
Fixed compilation on Linux, full case sensitive filenames and paths, more fixes

### PR DESCRIPTION
**Pull request by Skyrimus**
1) Fixed compilation on Linux
2) Compiled prebuilt bison for Linux x64
3) Compiled prebuilt flex for Linux x64
4) Compiled prebuilt ecc for Linux x64
5) Fixed afStarsPositions indexes
6) Fixed rendering map size on LoadingHook
7) Fixed paths in includes from \ slash to / 
8) Full case sensitive paths and filenames fixed
9) Changes some classes for normal build in windows and linux 
10) Fixed some float calculation
11) Fixed ShadowsMap in LightMixer (+ 5-6 fps)
12) Added new cheat Infinite Ammo
13) Fixed TFE and TSE maps rendering
14) Disabled smooth axes 
15) https://github.com/aarcangeli/Serious-Sam-Android/pull/45/commits/a75fbec0b719b0e1aa8247f3e6ca5b048a719a29

A lot of improves and code fixes... Just check commits

### CODE REVIEW

**Comment by DartPower** [comment]
Good work! 👍

**Comment by aarcangeli** [comment]
Hi @Skyrimus, I'm really excited about this PR.
Tell me when you're done, so I can make a full review.

**Comment by Skyrimus** [comment]
> Hi @Skyrimus, I'm really excited about this PR.
> Tell me when you're done, so I can make a full review.

Ok. i write you later

**Comment by aarcangeli** [comment]
@Skyrimus do not worry about failed checks, I'm still experiencing it

**Comment by Skyrimus** [comment]
> @Skyrimus do not worry about failed checks, I'm still experiencing it

Ok, now i fixing some lags

**Comment by Skyrimus** [comment]
@aarcangeli you can merge PR

**Comment by aarcangeli** [comment]
Why did you removed the x86 abi?

**Comment by Skyrimus** [comment]
@aarcangeli because x86 libs don t get same CRC as arm and crash network between arm and x86 with CRC error in diff, if use only arm libs this works fine on x86 devices and not get error, because x86 devices can run arm binaries by houdini arm emualtion

**Comment by aarcangeli** [comment]
> if use only arm libs this works fine on x86 devices and not get error, because x86 devices can run arm binaries by houdini arm emualtion

omg really? reallt cool.
Did you tried to compile also for 64bit arm?

**Comment by Skyrimus** [comment]
@aarcangeli, yes, tried, but it not works, maybe I destroyed something. You can try too?

**Comment by Skyrimus** [comment]
I think it is time to merge PR :D

**Comment by aarcangeli** [comment]
> @aarcangeli, yes, tried, but it not works, maybe I destroyed something. You can try too?

I don't think implementing 64 bits is a priority for two reasons:
- We don't have to put it on Google Play 
- I don't think it gain a fps boost



> I think it is time to merge PR :D

Wait, I'm testing

**Comment by Skyrimus** [comment]
> * I don't think it gain a fps boost

Hmm, maybe

File: Serious-Engine/Sources/Engine/Base/Anim.h [hunk 1]
```diff
@@ -23,10 +23,8 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #include <Engine/Base/Lists.h>
 
-#define NAME_SIZE 32
-typedef char NAME[NAME_SIZE];
-#define PATH_MAX 260
-typedef char FILE_NAME[PATH_MAX];
+typedef char NAME[32];
+typedef char FILE_NAME[260];
```
Comment (aarcangeli) [inline Serious-Engine/Sources/Engine/Base/Anim.h:27]:
```suggestion
#define NAME_SIZE 32
#ifndef PATH_MAX
#define PATH_MAX 260
#endif

typedef char NAME[NAME_SIZE];
typedef char FILE_NAME[PATH_MAX];
```
NAME_SIZE should not be removed or debug build will fail (it is referenced in assertions)

File: Serious-Engine/Sources/Engine/Ska/smcpars.cpp.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
+typedef union {
```
Comment (aarcangeli) [inline Serious-Engine/Sources/Engine/Ska/smcpars.cpp.h:1]: I think this file can be removed (it is unused)

File: Serious-Engine/Sources/Engine/Ska/smcPars.hpp [hunk 3]
```diff
@@ -0,0 +1,99 @@
+/* A Bison parser, made by GNU Bison 3.4.2.  */
```
Comment (aarcangeli) [inline Serious-Engine/Sources/Engine/Ska/smcPars.hpp:1]: I think this file can be removed (it is unused)

**Comment by Skyrimus** [comment]
On linux and windows it builds fine and full work

File: Serious-Engine/Sources/AndroidAdapters/android-adapters.h [hunk 4]
```diff
@@ -10,7 +10,21 @@
...
+#elif defined(__i386) || defined(__arm__)
+
+  #define PLATFORM_32BIT 1
+static_assert(sizeof(void *) == 4, "");
+
+#else
+  #error "Unknown CPU-Architecture, adapt this code to detect 32/64bitness of your system!"
+#endif
+
+static_assert(sizeof(size_t *) == sizeof(void *), "");
```
[the remark this answers is not in the document]
Comment (Skyrimus) [reply Serious-Engine/Sources/AndroidAdapters/android-adapters.h:27]: Ok

**Comment by Skyrimus** [comment]
Fixed now

**Comment by Skyrimus** [comment]
All good

**Pull request merged**